tests/extmod/machine_disable_irq.py

```diff
@@ -0,0 +1,26 @@
+# Test executing Python code while IRQs disabled.
+try:
+    from machine import disable_irq, enable_irq
+    from time import ticks_us, ticks_add, ticks_diff
+except ImportError:
+    print("SKIP")
+    raise SystemExit
+
+
+# Structured to also nest disable_irq() calls
+def f(n):
+    st_irq = disable_irq()
+    if n:
+        f(n - 1)
+    else:
+        # busy-wait in a tight loop for 1ms, to simulate doing some work in a critical section
+        tm_til = ticks_add(ticks_us(), 1000)
+        while ticks_diff(tm_til, ticks_us()) > 0:
+            pass
```

Comment:
I usually write this kind of pattern without using `ticks_add()` as:
```py
t0 = ticks_us()
while ticks_diff(ticks_us(), t0) < 1000:
    pass
```
I see that the way you've done it is the way the example is written in the `ticks_add()` docs... but you'll see in the `ticks_diff()` docs that it's done similar to what I wrote above.

Anyway, this is not really important, just FYI if you wanted to make these kinds of loops a little simpler.

Reply:
Yeah, I agree that's cleaner. Have changed, also rebased and re-ran tests on C3.